src/commands/topics.ts

```diff
@@ -460,3 +516,97 @@ export function registerTopicCommands(): vscode.Disposable[] {
+  includeKeySchema: boolean;
+  includeValueSchema: boolean;
+}> {
+  // TODO(shoup): update this when we migrate from ContainerTreeItem<Schema> to Subject modeling
+
+  // pre-pick any schema kinds that are already associated with this topic
+  const topicKeySubjects = subjectsToStrings(
+    topic.children.filter((subject) => ((subject.label as string) ?? "").endsWith("-key")),
+  );
+  const topicHasValueSchemas = subjectsToStrings(
+    topic.children.filter((subject) => ((subject.label as string) ?? "").endsWith("-value")),
+  );
+  const items: vscode.QuickPickItem[] = [
+    {
+      label: "Key Schema",
+      description:
+        topicKeySubjects.length > 0
+          ? `Associated subject(s): ${topicKeySubjects.join(", ")}`
+          : undefined,
+      detail: "Enable to select a schema for the message `key`",
+      picked: topicKeySubjects.length > 0,
+      iconPath: new vscode.ThemeIcon(IconNames.KEY_SUBJECT),
+    },
+    {
+      label: "Value Schema",
+      description:
+        topicHasValueSchemas.length > 0
+          ? `Associated subject(s): ${topicHasValueSchemas.join(", ")}`
+          : undefined,
+      detail: "Enable to select a schema the message `value`",
+      picked: topicHasValueSchemas.length > 0,
+      iconPath: new vscode.ThemeIcon(IconNames.VALUE_SUBJECT),
+    },
+  ];
+
+  const selectedItems = await vscode.window.showQuickPick(items, {
+    canPickMany: true,
+    title: `Producing to ${topic.name}: Select Schema Kind(s)`,
+    placeHolder: "Select which schema kinds to include (none for schemaless JSON)",
+    ignoreFocusOut: true,
+  });
+
+  return {
+    includeKeySchema: selectedItems?.some((item) => item.label === "Key Schema") ?? false,
+    includeValueSchema: selectedItems?.some((item) => item.label === "Value Schema") ?? false,
+  };
+}
+
+export function subjectsToStrings(subjects: ContainerTreeItem<Schema>[]): string[] {
+  return subjects.map((subject) => subject.label as string);
+}
+
+/**
+ * Prompt the user to select a schema to use when producing messages to a topic.
+ * @param topic The Kafka topic to produce messages to
+ * @param kind Whether this is for a 'key' or 'value' schema
+ * @returns The selected {@link Schema}, or `undefined` if user cancelled
+ */
+export async function promptForSchema(
+  topic: KafkaTopic,
+  kind: "key" | "value",
+): Promise<Schema | undefined> {
+  // look up any associated SR instance for this topic
+  const loader = ResourceLoader.getInstance(topic.connectionId);
+  const schemaRegistries: SchemaRegistry[] = await loader.getSchemaRegistries();
+  const defaultRegistry: SchemaRegistry | undefined = schemaRegistries.find(
+    (registry) => registry.environmentId === topic.environmentId,
+  );
+
+  // prompt for the Schema Registry to use, highlighting the default registry for this topic
+  // (if it exists)
+  const registry: SchemaRegistry | undefined = await schemaRegistryQuickPick(
```

Comment:
TODO: pre-pick / guard against SR with different connection ID than the topic until the sidecar can support it

Reply:
Done in https://github.com/confluentinc/vscode/pull/1093/commits/d563d1146d8c98c6db20275cf8a28173669dc1a6